R/utils.R

```diff
@@ -167,6 +167,10 @@ rss = function() {  #5515 #5517
 }
 
 frev = function(x, copy=TRUE) {
-    ans = if (isAltrep(x)) last(x):first(x) else .Call(Cfrev, x, copy)
-    if (isTRUE(copy)) ans else invisible(ans)
+    if (!isTRUEorFALSE(copy))
+      stopf("'copy' must be TRUE or FALSE.")
+    altrep = isAltrep(x)
+    if (altrep && !copy) warning("Given vector is altrep and will not be modified in-place.")
```

Comment:
We cannot accept such unpredictable behavior. It may happen that a workflow (that depends on update in place) will work fine for a month, and then suddenly getting another type (altrep) of input it will fail.

Reply:
Good call. We also do not know how `first/last` might be dispatched, hence, relying on them seems error-prone.

Reply:
relevant broader issue #4697. I would approach the broader issue separately from this PR. we might plan out a considered approach to adding ALTREP that minimizes breakage risk.